### PR TITLE
FEATURE: Locale support for seeded categories and topics

### PR DESCRIPTION
**Pull request by gschlager**
* Stores IDs of seeded topics as hidden site settings and converts the `is_welcome_topic` custom field into a site setting as well.

* Makes the topic title of the Admin Quick Start Guide translatable. The guide's content not yet.

* Adds a new rake task `i18n:reseed` to update seeded topics and categories with the latest translations.
(Works only for newly created sites which have the new hidden site settings or when the DB migration was able to identify existing topics.)

* Deduplicates code used by `db/fixtures/990_topics.rb` and the now obsolete `topics:update_static` rake task.

* Doesn't create the Lounge, Staff and Site Feedback categories in a migration anymore. That's now happening in the seeding phase.

* Adds a new dialog for reseeding categories and topics that can be accessed via a new button in Customize / Text Content.

  image

  image

* The dialog shows only categories and topics that could be found. It preselects topics and categories when the content hasn't been changed by a user yet.

* The wizard automatically reseeds unchanged topics and categories when the locale is changed.

* The `default_locale` site setting has a link to Customize / Text Content.
  
  image

### CODE REVIEW

File: db/fixtures/001_refresh.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# fix any bust caches post initial migration
+ActiveRecord::Base.send(:subclasses).each { |m| m.reset_column_information }
+SiteSetting.refresh!
```
Comment (gschlager) [inline db/fixtures/001_refresh.rb]: Those 2 lines is the only remaining code from `db/fixtures/001_categories.rb`. I guess it's still needed?

Comment (ZogStriP) [reply db/fixtures/001_refresh.rb]: I think both are still needed, yes.

File: app/assets/javascripts/admin/controllers/modals/admin-reseed.js.es6 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  loading: true,
+  reseeding: false,
+  categories: null,
+  topics: null,
+
+  onShow() {
+    ajax("/admin/customize/reseed")
+      .then(result => {
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/controllers/modals/admin-reseed.js.es6]:
```
      .then(result => 
        this.setProperties({categories: result.categories, topics: result.topics})
      )
```

File: app/assets/javascripts/admin/templates/site-text-index.hbs [hunk 3]
```diff
@@ -8,11 +8,19 @@
                key-up=(action "search")}}
 
   <div class='extra-options'>
+    {{d-button action=(route-action "showReseedModal")
+               class="btn btn-default"
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/templates/site-text-index.hbs]: dont think btn is necessary here

File: app/assets/javascripts/admin/templates/modal/admin-reseed.hbs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {{topic.name}}
+          </label>
+        {{/each}}
+      </fieldset>
+    {{/if}}
+  {{/conditional-loading-spinner}}
+{{/d-modal-body}}
+
+<div class="modal-footer">
+  {{d-button action=(action "reseed") class="btn btn-danger" label="go_ahead" disabled=reseeding}}
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/templates/modal/admin-reseed.hbs]: btn not needed I think

File: app/assets/javascripts/admin/templates/modal/admin-reseed.hbs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{{#d-modal-body title="admin.reseed.modal.title" subtitle="admin.reseed.modal.subtitle" class="reseed-modal"}}
+  {{#conditional-loading-spinner condition=loading}}
+    {{#if categories}}
+      <fieldset>
+        <legend>{{i18n "admin.reseed.modal.categories"}}</legend>
+
+        {{#each categories as |category|}}
+          <label>
+            {{input type="checkbox" checked=category.selected}}
+            {{category.name}}
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/templates/modal/admin-reseed.hbs]: can you wrap this in a span please

File: app/assets/javascripts/admin/templates/modal/admin-reseed.hbs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    <br>
+
+    {{#if topics}}
+      <fieldset>
+        <legend>{{i18n "admin.reseed.modal.topics"}}</legend>
+
+        {{#each topics as |topic|}}
+          <label>
+            {{input type="checkbox" checked=topic.selected}}
+            {{topic.name}}
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/templates/modal/admin-reseed.hbs]: can you wrap this in a span please

File: app/assets/stylesheets/common/admin/customize.scss [hunk 7]
```diff
@@ -781,3 +781,15 @@
     }
   }
 }
+
+.reseed-modal {
+  legend {
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/admin/customize.scss]: as much as possible I avoid styling directly on html elements

File: lib/seed_data/topics.rb [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+          category: staff_category
+        }
+      end
+
+      topics
+    end
+
+    def create_topic(site_setting_name:, title:, raw:, category: nil, static_first_reply: false, after_create: nil)
+      topic_id = SiteSetting.send(site_setting_name)
+      return if topic_id > 0 || Topic.find_by(id: topic_id)
```
Comment (gschlager) [inline lib/seed_data/topics.rb]: Our current topic seeding doesn't recreate missing ToS, FAQ and Privacy Policy topics and I kept that behavior. Should we force the creation of those topics if they somehow got deleted and the site settings point to non-existent topics? We had such a case a week ago: https://meta.discourse.org/t/how-to-regenerate-faq-and-tos-pages/110625

**Pull request merged**